torchdynamo/codegen.py

```diff
@@ -110,6 +118,18 @@ def __call__(self, value, allow_cache=True):
                 self._create_load_const(graph_outputs[graph_outputs_key].index)
             )
             output.append(create_instruction("BINARY_SUBSCR"))
+            if isinstance(value, UnspecializedPrimitiveVariable):
+                output.extend(
+                    [
+                        self.create_load_attr("numpy"),
+                        create_instruction("CALL_FUNCTION", 0),
+                        self.create_load_attr("reshape"),
+                        self._create_load_const(1),
+                        create_instruction("CALL_FUNCTION", 1),
+                        self._create_load_const(0),
+                        create_instruction("BINARY_SUBSCR"),
+                    ]
```

Comment:
Agree that converting back and forth is not guaranteed to preserve types, actually [this](https://github.com/pytorch/torchdynamo/pull/392#discussion_r896211756) is one of the discrepancies I hit.
But I still don't think adding a property is the best solution, as we put the unspecialized into ```fx.Graph```, so it may go through many ops from input to output. If we add any property, we have to define all these rules explicitly. E.g, ```np.int32 / int = np.float64```. I think we wrap them as 1-element tensor is to leverage the facility the tensor ops provided. Correct me if I'm wrong.

Reply:
Update: it seems there is a way to do value/type propagation. If unspec ops tensor, it will return tensor and don't need the original type to unwrap; if unspec ops unspec/const, it will return unspec and need the original type for unwrap. So I made change in ```BuiltinVariable.call_function``` to calculate the value during compiling and store the result as ```raw_value``` in unspec variable. The downside is max/min is a bit ugly as it needs special handling. Let me know if this looks good to you. Thank you!